test/unit/player.test.js

```diff
@@ -1428,59 +1428,22 @@ QUnit.test('Player#tech logs a warning when called without a safety argument', f
 });
 
 QUnit.test('player#reset loads the Html5 tech and then techCalls reset', function(assert) {
-  let loadedTech;
-  let loadedSource;
-  let techCallMethod;
-
-  const testPlayer = {
-    options_: {
-      techOrder: ['html5', 'flash']
-    },
-    resetCache_() {},
-    loadTech_(tech, source) {
-      loadedTech = tech;
-      loadedSource = source;
-    },
-    techCall_(method) {
-      techCallMethod = method;
-    },
-    resetControlBarUI_() {},
-    poster() {}
-  };
-
-  Player.prototype.reset.call(testPlayer);
-
-  assert.equal(loadedTech, 'html5', 'we loaded the html5 tech');
-  assert.equal(loadedSource, null, 'with a null source');
-  assert.equal(techCallMethod, 'reset', 'we then reset the tech');
-});
+  this.clock.restore();
+  const done = assert.async();
```

Comment:
we try to avoid making tests async if we can.

Reply:
I have to disagree here. The previous test was heavily stubbing the player object as well as relying a particular internal API. What I'm most concerned about, though, is the fact that the test was forcing the internal methods of `reset` to be called, even though this code wouldn't work in a prod environment. In prod, the player needs to be `ready` before we can interact with the tech. The point is, if we are forcing behavior to work in tests, then we can't rely on these tests to make sure that our production code will work as expected.